library/Odesk/Phystrix/AbstractCommand.php

```diff
@@ -280,6 +279,23 @@ public function setServiceLocator(LocatorInterface $serviceLocator)
         $this->serviceLocator = $serviceLocator;
     }
 
+
+    /**
+     * Records list of actions that have taken place, with their exceptions
+     *
+     * @param string    $eventName  type from class constants EVENT_*
+     * @param Exception $e          exception that was thrown as part of $eventName
+     */
+    protected function handleEvent($eventName, \Exception $e = null )
```

Comment:
@lcf  IMHO, the event and exception are actually tied together (one drove the other). This function allows for all event-related actions that need to take place (such as setting the exception, name) to be combined in a single location. In our downstream implementation, we then trigger an event based on all this information